File: bot.js
const Discord = require('discord.io');
const config = require("./config.json");
const secrets = require("./secrets.json");
const request = require('request')

console.log("Starting bot...");

const client = new Discord.Client({
    token: secrets.token,
    autorun: true
});

client.on("ready", () => {
    console.log("bot has started!");
});

client.on("message", (message) => {
    if (message.author.bot) return;

    if (message.content.indexOf(config.prefix) !== 0) return;

    const args = message.content.slice(config.prefix.length).trim().split(/ +/g);
    const command = args.shift().toLowerCase();

    if (command === "gen") {
        request(secrets.apiUrl, { json: true }, (err, res, body) => {
            if (err) { return console.log(err); }
            message.delete().catch(O_o=>{});
            message.channel.send(body.sampleDesc);
        })
    }
    else {
        message.delete().catch(O_o=>{});
        message.channel.send("did not recognise your request! sorry!");
    }
});

console.log("Bot has logged in!");

File: bot.py
import json
import discord
import asyncio
import requests

config = json.loads(open('secrets.json').read())

TOKEN = config["token"]

client = discord.Client()

@client.event
async def on_message(message):
    print('Checking if I\'m needed...')
    if message.author == client.user:
        return
    if message.content == '-npc':
        print('MAKING AN NPC Y\'ALL!')
        res = requests.get(config["apiUrl"])
        print(res.json())
        body = json.loads(res.json())
        await client.send_message(message.channel, body["sampleDesc@"])

@client.event
async def on_ready():
    print('Logged in as')
    print(client.user.name)
    print(client.user.id)
    print('--------')

client.run(TOKEN, bot=True)